### ISSUE TITLE
large memory requirement and getting low accuracy after adding adapter layers

### ISSUE THREAD

**Issue by rabeehk** [adapter-hub/adapter-transformers#93]
Hi
I am using a custom model, so I had to read and implement the adapter layers using library myself, here is how I did it:
I defined an adapter class like below:

```
"""Implements an Adapter block.

Code is adapted from: https://github.com/Adapter-Hub/adapter-transformers/blob/master/\
src/transformers/adapter_modeling.py
"""
import torch.nn as nn
from .adapter_utils import Activations

class Adapter(nn.Module):
    """
    Implementation of a single Adapter block.
    """

    def __init__(self, input_size, config):
        super().__init__()
        self.input_size = input_size
        self.add_layer_norm_before = config.add_layer_norm_before
        self.add_layer_norm_after = config.add_layer_norm_after
        self.residual_before_layer_norm = config.residual_before_layer_norm

        # list for all modules of the adapter, passed into nn.Sequential()
        seq_list = []
        # If we want to have a layer norm on input, we add it to seq_list
        if self.add_layer_norm_before:
            seq_list.append(nn.LayerNorm(self.input_size))

        # if a downsample size is not passed, we just half the size of the original input
        reduction_factor = config.reduction_factor if config.reduction_factor is not None else 2
        self.down_sample_size = self.input_size//reduction_factor
        seq_list.append(nn.Linear(self.input_size, self.down_sample_size))
        self.non_linearity = Activations(config.non_linearity.lower())
        seq_list.append(self.non_linearity)

        # sequential adapter, first downproject, then non-linearity then upsample.
        # In the forward pass we include the residual connection
        self.adapter_down = nn.Sequential(*seq_list)

        # Up projection to input size
        self.adapter_up = nn.Linear(self.down_sample_size, self.input_size)

        # If we want to have a layer norm on output, we apply it later after a
        # separate residual connection. This means that we learn a new output layer norm,
        # which replaces another layer norm learned in the bert layer
        if self.add_layer_norm_after:
            self.adapter_norm_after = nn.LayerNorm(self.input_size)

    def forward(self, x): #, residual_input):
        down = self.adapter_down(x)
        up = self.adapter_up(down)
        output = up

        # todo add brief documentation what that means
        #if self.residual_before_layer_norm:
        #    output = output + residual_input

        # todo add brief documentation what that means
        if self.add_layer_norm_after:
            output = self.adapter_norm_after(output)

        # todo add brief documentation what that means
        #if not self.residual_before_layer_norm:
        #    output = output + residual_input

        return output #, down, up
```

then I add them between the layers of my class

```
class LayerFF(nn.Module):
    def __init__(self, config):
        super().__init__()
        self.DenseReluDense = T5DenseReluDense(config)
        self.layer_norm = T5LayerNorm(config.d_model, eps=config.layer_norm_epsilon)
        self.dropout = nn.Dropout(config.dropout_rate)
        # TODO: remove it later.
        self.add_adapters = config.add_adapters
        if self.add_adapters:
            # TODO: adapter config should be a part of model config then. or do we want a separate config?
            adapter_config = AdapterConfig()
            self.adapter = Adapter(config.d_model, adapter_config)
    def forward(self, hidden_states):
        norm_x = self.layer_norm(hidden_states)
        y = self.DenseReluDense(norm_x)
        if self.add_adapters:
            y = self.adapter(y)
        layer_output = hidden_states + self.dropout(y)
        return layer_output
```
then I freeze all model parameters with require_grad=False and setting only adapter ones to True. I see a large memory requirement, around the same model, could you assist me if anything is missing? thanks

I additionally get accuracy close to untrained model, and very low, could you give me possible suggestions to improve the accuracy? thank you.

**Comment by adapter-hub-bert** [comment]
This issue has been automatically marked as stale because it has been without activity for 90 days. This issue will be closed in 14 days unless you comment or remove the stale label.

**Comment by adapter-hub-bert** [comment]
This issue was closed because it was stale for 14 days without any activity.
